Tests/_CryptoExtrasTests/ChaCha20CTRTests.swift

```diff
@@ -0,0 +1,99 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import Foundation
+import XCTest
+import Crypto
+import _CryptoExtras
+
+class ChaCha20CTRTests: XCTestCase {
+
+    /// Test Vector - https://datatracker.ietf.org/doc/html/rfc9001#name-chacha20-poly1305-short-hea
+    func testChaCha20CTR_v1() throws {
+        let hpKey = try Array(hexString: "25a282b9e82f06f21f488917a4fc8f1b73573685608597d0efcb076b0ab7a7a4")
+        /// Sample = 0x5e5cd55c41f69080575d7999c25a5bfb
+        let counterAsData = try Array(hexString: "5e5cd55c")
+        let counterAsUInt32 = counterAsData.withUnsafeBytes { $0.load(as: UInt32.self) }
```

Comment:
Rather than use a load, can we write the integer hex literal here? That avoids having symmetric bugs in the loading and product code.

Reply:
In order to keep the hex orientation I needed to use UInt32(bigEndian:) initializer. Is that alright? Or would you like me to convert the hex into it's actual decimal value and use that instead?

Reply:
```swift
let counterAsUInt32 = UInt32(bigEndian: 0x5e5cd55c)
// or
let counterAsUInt32 = UInt32(1557486686)
```